tests/integration/conftest.py

```diff
@@ -317,9 +318,19 @@ def get_relation_data(
     return RelationData(provider=provider_data, requirer=requirer_data)
 
 
-def assert_can_ping(ip, port):
-    response = os.system(f"ping -c 1 {ip} -p {port}")
-    assert response == 0, f"{ip}:{port} is down/unreachable"
+def _can_connect(ip, port) -> bool:
+    try:
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.connect(ip, int(port))
+        return True
+    except OSError:
```

Comment:
There are [more kinds](https://docs.python.org/3/library/socket.html#socket.socket.connect). For itests perhaps we can have a bare `except`?

Reply:
Good point, changed to a bare `except`.